snowfakery/standard_plugins/Salesforce.py

```diff
@@ -330,26 +330,14 @@ def _load_dataset(self, iteration_mode, rootpath, kwargs):
                 f"Unable to query records for {query}: {','.join(qs.job_result.job_errors)}"
             )
 
-        self.tempdir, self.iterator = create_tempfile_sql_db_iterator(
+        tempdir, iterator = create_tempfile_sql_db_iterator(
             iteration_mode, fieldnames, qs.get_results()
         )
-        return self.iterator
+        iterator.cleanup.push(tempdir)
```

Comment:
The temporary directory object needs to live a bit longer than the local tempfile, which needs to live a bit longer than than the SQL Alchemy connection to the tempfile.

So I use an ExitStack to inject cleanup code which is run before the connection is closed and the file handle is released.

This line adds to the ExitStack.

All of this is triggered by the garbage collector letting go of the iterator when the recipe is finishing up.

It might be better to of a deterministic, explicit, top-down cleanup, but that should probably be a different PR.